test/client/gateways/Gateway.test.ts

```diff
@@ -1,7 +1,64 @@
-import {fakeRequest, send, TIMEOUT_ERROR, UNKNOWN_ERROR} from '../../../src/client/gateways/Gateway'
+import {fakeRequest, send, TIMEOUT_ERROR, UNKNOWN_ERROR,
+  post, put, patch, get} from '../../../src/client/gateways/Gateway'
 import {Request} from 'superagent'
 
 describe('Gateway', () => {
+  const url = "http://dummy-server.com/order";
+  const body = {orderId: 1};
+  const responseTimeOut = 30000;
+  const timeOut = 60000;
+
+  describe('request types', () => {
+    describe('post', () => {
+      const request = post(url, body) as any;
+
+      expect(request.url).toEqual(url)
+      expect(request.method).toEqual("POST")
+      expect(request._data).toEqual(body)
+      expect(request.header['Accept']).toEqual("application/json; charset=utf-8")
+      expect(request.header['Content-Type']).toEqual("application/json; charset=utf-8")
+      expect(request._responseTimeout).toEqual(responseTimeOut)
+      expect(request._timeout).toEqual(timeOut)
+      expect(request._maxRetries).toEqual(1)
+    });
```

Comment:
Having to cast this to `any` is a bit of a smell and should have given a lint warning (which I think I'll make an error in future, forcing people to suppress if they really need any). This looks like we are testing some internal properties (pretty sure using underscore is a convention for private methods/properties), I'm not convinced we should be doing that 🤔

Reply:
@Hi @GentlemanHal , I second the thought where you say that we shouldn't be relying on implementation. Completely agree on that. but I am wondering how can we test whether `Request` is being configured the right way, with right `retry`, `timeout` and `body` parameters?

Reply:
Hi @GentlemanHal , I've removed those assertions now which was testing private fields such as `_data`, `_responseTimeout`, `_timeout` and `_maxRetries` and only testing which are publicly available. Let me know if that works for you.

Reply:
It's a tricky one, as making sure we don't make silly typos 😳makes sense. Perhaps the real answer is a higher level integration or functional test? Although having a test that has to wait a minute also doesn't sit well with me 🤔 

I don't expect you to do anything extra for this pull request, I'm mostly just thinking out loud 😄

Reply:
Apologies, somehow I ignored the fact that **Request** is a promise over resposne and will take time to run them. my bad, i'll keep that in mind from the next time

Reply:
Ah sorry if I wasn't clear, the unit tests you wrote were fine and complete quickly (17ms)

I was just suggesting to really test the timeout stuff, we probably need a functional test that literally tries to connect to a slow server and waits the full 1 minute until everything times out as expected.